frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/ClickBehaviorSidebar.jsx

```diff
@@ -16,188 +19,208 @@ import TableClickBehaviorView from "./TableClickBehaviorView";
+      setOriginalColumnVizSettings(originalColumnVizSettings);
+    },
+    [dashcard],
+  );
+
+  const handleUnsetSelectedColumn = useCallback(() => {
+    if (!isValidClickBehavior) {
+      handleChangeSettings(originalColumnVizSettings);
     }
-    this.setState({ originalColumnVizSettings: null, selectedColumn: null });
-  };
-
-  getClickBehavior(props = this.props) {
-    const { dashcard } = props;
-    const { selectedColumn } = this.state;
-    if (isTableDisplay(dashcard) && selectedColumn == null) {
-      return undefined;
+    setOriginalColumnVizSettings(null);
+    setSelectedColumn(null);
+  }, [isValidClickBehavior, originalColumnVizSettings, handleChangeSettings]);
+
+  const handleCancel = useCallback(() => {
+    onReplaceAllDashCardVisualizationSettings(dashcard.id, originalVizSettings);
+    hideClickBehaviorSidebar();
+  }, [
+    dashcard,
+    originalVizSettings,
+    hideClickBehaviorSidebar,
+    onReplaceAllDashCardVisualizationSettings,
+  ]);
+
+  useOnMount(() => {
+    if (shouldShowTypeSelector(clickBehavior)) {
+      setTypeSelectorVisible(true);
     }
-    if (selectedColumn == null) {
-      return getIn(dashcard, ["visualization_settings", "click_behavior"]);
-    } else {
-      return this.getClickBehaviorForColumn(props, selectedColumn);
-    }
-  }
-
-  getClickBehaviorForColumn(props, column) {
-    return getIn(props.dashcard, [
-      "visualization_settings",
-      "column_settings",
-      keyForColumn(column),
-      "click_behavior",
-    ]);
-  }
-
-  getColumns() {
-    const { dashcard, dashcardData } = this.props;
-    return getIn(dashcardData, [dashcard.card_id, "data", "cols"]);
-  }
-
-  showTypeSelectorIfNeeded() {
-    const { type } = this.getClickBehavior() || {};
-    this.setState({ showTypeSelector: type == null });
-  }
-
-  updateSettings = (
-    click_behavior,
-    { props = this.props, state = this.state } = {},
-  ) => {
-    const { selectedColumn } = state;
-    const { id } = props.dashcard;
-    if (selectedColumn == null) {
-      props.onUpdateDashCardVisualizationSettings(id, { click_behavior });
-    } else {
-      props.onUpdateDashCardColumnSettings(id, keyForColumn(selectedColumn), {
-        click_behavior,
-      });
+    if (dashcard) {
+      setOriginalVizSettings(dashcard.visualization_settings);
     }
-  };
+  });
 
-  handleCancel = () => {
-    this.props.onReplaceAllDashCardVisualizationSettings(
-      this.props.dashcard.id,
-      this.state.originalVizSettings,
-    );
-    this.props.hideClickBehaviorSidebar();
-  };
-
-  render() {
-    const { dashboard, dashcard, parameters, hideClickBehaviorSidebar } =
-      this.props;
-    const { selectedColumn } = this.state;
-
-    const clickBehavior = this.getClickBehavior() || { type: "menu" };
+  useEffect(() => {
+    if (!previousDashcard) {
+      return;
+    }
 
-    if (isTableDisplay(dashcard) && selectedColumn == null) {
+    if (dashcard.id !== previousDashcard.id) {
```

Comment:
Maybe by making this line

`if (dashbard.id !== previousDashcard?.id) {`

we can remove the `if` guard above this `if` block.